Support for XCode 16

troysandal opened the pull request:
I've gotten almost all the examples and tools working in XCode 16 using modern XCFrameworks and updated Pods.  Remove TextMate support and fixed one Wax Unit Tests. The example apps I ported run on iOS18 but that's all I've tested.  I ran the WAX_TEST in an example and got 8 errors, was able to fix one simple one.

Not Done
- Tested Wax XCode Template
- Updated Twitter example - X killed the API it's using, should either refactor or drop this example.
- Lua 5.1.5 Update
- Lua 5.x Support

troysandal committed:
Tools xcodeproject upgrade to XCode 16

troysandal committed:
fix wax structs tests

wasn't checking for 64 bit arch and using doubles instead of floats.

troysandal committed:
import block methods vs inline decls

troysandal committed:
updated for iOS 13+

troysandal committed:
more helpful debug statements

troysandal committed:
don't run unit tests by default

troysandal committed:
Framework now built as modern XCFramework

troysandal committed:
BlockExample working in XCode 16

troysandal committed:
MapKitExample runs in XCode 16

troysandal committed:
mobdebug built as XCode 16 XCFramework

troysandal committed:
LuaCodeDebug ZeroBrane Debugging works XCode 16

troysandal committed:
remove bad ignore, these are actually needed

troysandal committed:
States building in XCode 16 using wax pod

troysandal committed:
add missing xcode project and workspace files

not sure why junzhan added these to the .gitignore, they are definitely needed except the *.xcuserdata which was already covered

troysandal committed:
use new name of wax xc framework

troysandal committed:
AutoTest working in XCode 16

Did not checkin changes to wax.podspec to support this as it exposes more of lua/tolua++ and I'm not sure it's warranted (wasn't already done by junzhan).  Also replaced NSFW image.

troysandal committed:
WatchFramework compilng for XCode 16

troysandal committed:
Twitter app building on XCode 16

Won't run because X (the company formerly known as Twitter) changed their API.

troysandal committed:
Remove TextMate support

RIP old friend.

troysandal committed:
IBOutlet working in Xocde 16

troysandal committed:
all pods require iOS 15 or greater

troysandal committed:
States (no framework) running in XCode 16

troysandal committed:
Install via Pod working in XCode 16

troysandal committed:
Install with Source working in XCode 16

troysandal committed:
Install with Framework running in XCode 16

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/alibaba/wax?pullRequest=68) <br/>All committers have signed the CLA.

troysandal committed:
unused imports

troysandal committed:
initial Luau support

troysandal committed:
fix Strict Prototypes compiler warnings

troysandal committed:
more helpful enums

troysandal committed:
cleanup Luau path macros

troysandal committed:
incorporate fix from BowerHaus

https://github.com/bowerhaus/CoreyJohnsonWax/commit/2c4694099231224f098816093197244e2f1acca6

troysandal committed:
Use minimum supported platform

troysandal committed:
Luau Support

To keep the wax codebase compilng for both Lua 5.x and Luau I added a shim file that redefines Lua 5.x macros to be compatible with Luau.  It's invasive and not my favorite solution but it seems least invasive of all fixes.

Updated all examples & tools.

troysandal committed:
Luau - use private __env vs setfenv

troysandal committed:
Luau - use private __env vs setfenv

troysandal committed:
Merge branch 'master' of github.com:troysandal/wax

troysandal committed:
bug fix - we left env tabel on the stack

troysandal committed:
Moved environments into a weak, named table

We were trying to set the environment on the metatable of the userdata but didn't realize they all share the same metatable.  Now stored in a weak registry table.  And it works?

troysandal committed:
Remove setfenv luaL_ref code

Gideros resets the Registry between runs of the player but we aren't notified when this happens and our ref goes bad.  If we could figure this out we can move back to refs which are much faster.

troysandal committed:
fix compile error in lua 5.1

troysandal committed:
file should have never been checked in

troysandal committed:
Note that Swift project is not upgradeable.

troysandal committed:
Migrate to XCode 16

troysandal committed:
more 5.1 compilation errors

troysandal committed:
syncn with AWB fixes

troysandal committed:
Luau Crash in methodClosure
- lua_pointer() impl changed
- https://github.com/troysandal/wax/issues/1

troysandal committed:
Remove non working tests

Thes tests relied upon the return value from retainCount which is not defined and has changed since the tests were written.

troysandal committed:
Fix broken unit tests

init methods should have always return self

troysandal committed:
omit non-helper files

We should consider deleting these in the future.

troysandal committed:
convert to modern lua modele

troysandal committed:
io not working in Luau, needs better fix

troysandal committed:
dont' assume init is implicitly looked for

troysandal committed:
explanatory comment

troysandal committed:
fix wax_luau abs folder

troysandal committed:
remove last abs path

troysandal committed:
Unit Test Failure
- https://github.com/troysandal/wax/issues/5

andy380743909 commented:
Hi bro,

Have you ever tried to upgrade to Lua 5.3 or 5.4?

troysandal commented:
> Have you ever tried to upgrade to Lua 5.3 or 5.4?

Out of scope for my project.  Lua > 5.1 removed lua_setfenv()/lua_getfenv() and host of other things that would make that difficult. Roblox did similar things but I was able to hack around it.  You're welcome to try it yourself if you'd like and I'd be happy to review your work.

andy380743909 commented:
> > Have you ever tried to upgrade to Lua 5.3 or 5.4?
> 
> Out of scope for my project. Lua > 5.1 removed lua_setfenv()/lua_getfenv() and host of other things that would make that difficult. Roblox did similar things but I was able to hack around it. You're welcome to try it yourself if you'd like and I'd be happy to review your work.

https://github.com/andy380743909/wax/discussions/1

I have done some works about upgrading to Lua 5.3 5.4, but I'm not very sure whether it's correct. Please have a look, thanks very much!

troysandal commented:
I don't have time this week but can look at it next week.
